cumulus_library/template_sql/utils.py

```diff
@@ -0,0 +1,131 @@
+"""utility functions related to populating jinja templates
+
+Currently, these deal with various edge cases around complex nested objects in sql
+and validating them so that the actual generated queries can be constructed more
+simply. This includes, but is not limited, to:
+    - Data missing entirely
+    - Data present, but 'nullish' - with some structure but no value
+    - Data with deep missing elements
+    - Data which may or may not be in an array depending on context
+"""
+from rich.progress import Progress, Task
+
+from cumulus_library.template_sql.templates import (
+    get_column_datatype_query,
+    get_is_table_not_empty_query,
+)
+
+
+def is_codeable_concept_populated(
+    schema: str,
+    table: str,
+    base_col: str,
+    cursor,
+    progress: Progress,
+    task: Task,
+    coding_element="coding",
+) -> bool:
+    """Check db to see if codeableconcept data exists.
+
+    Will execute several exploratory queries to see if the column in question
+    can be queried naively. Will advance the associated progress's task by 3 steps.
```

Comment:
There is a lot of implicit task total tracking happening - I don't have a better solution right now, but I'm registering my unease. As I add more progress bars to the ETL, prepare yourself to mock me when I do the same thing.

(One possible thought - at this point, trade precision-of-bar for maintainability and have this just advance 1 and/or bring the advancing up a layer to the caller who advances 1)

Reply:
yeah i don't love it either - it could be a constant that's imported? but then we get into `total=len(dict)*utils.ADVANCE_NUM+2` or something.

I suppose in this case - we could consider this an 'atomic' operation and just advance once in the caller